v2/codetags/parser_test.go

```diff
@@ -29,100 +30,462 @@ func TestParse(t *testing.T) {
-			{Name: "s", Value: "value", Type: ArgTypeString},
-			{Name: "i", Value: "2", Type: ArgTypeInt},
-			{Name: "b", Value: "true", Type: ArgTypeBool},
-		}},
+		// Basic tag name tests
+		{
+			name:   "simple name",
+			input:  "name",
+			expect: mkt("name"),
+		},
+		{
+			name:   "name with comment",
+			input:  "name // comment",
+			expect: mkt("name"),
+		},
+		{
+			name:   "name with dash",
+			input:  "name-dash",
+			expect: mkt("name-dash"),
+		},
+		{
+			name:   "name with dot",
+			input:  "name.dot",
+			expect: mkt("name.dot"),
+		},
+		{
+			name:   "name with colon",
+			input:  "name:colon",
+			expect: mkt("name:colon"),
+		},
+
+		// String arguments tests
+		{
+			name:   "empty parentheses",
+			input:  "name()",
+			expect: mkt("name"),
+		},
+		{
+			name:   "single argument",
+			input:  "name(arg)",
+			expect: mkta("name", mksa("arg")),
+		},
+		{
+			name:   "argument with comment",
+			input:  "name(arg) // comment",
+			expect: mkta("name", mksa("arg")),
+		},
+		{
+			name:   "uppercase argument",
+			input:  "name(ARG)",
+			expect: mkta("name", mksa("ARG")),
+		},
+		{
+			name:   "mixed case argument",
+			input:  "name(ArG)",
+			expect: mkta("name", mksa("ArG")),
+		},
+		{
+			name:   "argument with dash",
+			input:  "name(has-dash)",
+			expect: mkta("name", mksa("has-dash")),
+		},
+		{
+			name:   "argument with dot",
+			input:  "name(has.dot)",
+			expect: mkta("name", mksa("has.dot")),
+		},
+		{
+			name:   "lowercase argument",
+			input:  "name(lower)",
+			expect: mkta("name", mksa("lower")),
+		},
+		{
+			name:   "uppercase argument",
+			input:  "name(CAPITAL)",
+			expect: mkta("name", mksa("CAPITAL")),
+		},
+		{
+			name:   "mixed case argument 1",
+			input:  "name(MiXeD)",
+			expect: mkta("name", mksa("MiXeD")),
+		},
+		{
+			name:   "mixed case argument 2",
+			input:  "name(mIxEd)",
+			expect: mkta("name", mksa("mIxEd")),
+		},
+		{
+			name:   "underscore in argument",
+			input:  "name(_under)",
+			expect: mkta("name", mksa("_under")),
+		},
+		{
+			name:   "quoted argument",
+			input:  `name("hasQuotes")`,
+			expect: mkta("name", mksa("hasQuotes")),
+		},
+		{
+			name:   "raw quoted argument",
+			input:  "name(`hasRawQuotes`)",
+			expect: mkta("name", mksa("hasRawQuotes")),
+		},
+		{
+			name:   "raw string with spaces",
+			input:  "withRaw(`a = b`)",
+			expect: mkta("withRaw", mksa("a = b")),
+		},
+
+		// Error cases for arguments
+		{
+			name:  "space in argument",
+			input: "name(has space)",
+			err:   true,
+		},
+		{
+			name:  "multiple comma-separated args",
+			input: "name(multiple, args)",
+			err:   true,
+		},
+		{
+			name:  "unclosed parenthesis",
+			input: "name(noClosingParen",
+			err:   true,
+		},
+		{
+			name:  "comma-separated args",
+			input: "name(arg1, arg2)",
+			err:   true,
+		},
+		{
+			name:  "unclosed raw string",
+			input: "badRaw(missing`)",
+			err:   true,
+		},
+		{
+			name:  "mixed arg formats",
```

Comment:
not sure what this is proving?

Just making it "err: true" means you can't see WHY it failed.  Maybe substring match?

Reply:
Dropped this test.  Added substring error matches.